core/commands/storage.go

```diff
@@ -253,6 +251,35 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 	Type: UploadRes{},
 }
 
+func controlSessionTimeout(ss *storage.Session) {
+	// error is special std flow, will not be counted in here
+	for curStatus := 0; curStatus < len(storage.StdSessionStateFlow)-1; {
+		select {
+		case sessionState := <-ss.TimeOutChan:
+			if sessionState.Succeed {
+				curStatus = sessionState.CurrentStep + 1
+			} else {
+				ss.SetStatus(storage.ErrStatus)
```

Comment:
If error status is set, this goroutine should return, it should not continue this for loop.